### PR TITLE
Fix Fix crash in RouteStepProgress.upcomingIntersection

### PR DESCRIPTION
**Pull request by kried**
### Description
Fixes possible incorrect index for empty intersections.

### CODE REVIEW

**Review by ShanMa1991 — APPROVED** [review]

File: Sources/MapboxCoreNavigation/RouteStepProgress.swift [hunk 1]
```diff
@@ -84,7 +84,8 @@ open class RouteStepProgress: Codable {
      The step must contain `intersectionsIncludingUpcomingManeuverIntersection` otherwise this property will be `nil`.
      */
     public var upcomingIntersection: Intersection? {
-        guard let intersections = intersectionsIncludingUpcomingManeuverIntersection, intersections.startIndex..<intersections.endIndex-1 ~= intersectionIndex else {
+        guard let intersections = intersectionsIncludingUpcomingManeuverIntersection, intersections.count > 0,
```
Comment (MaximAlien) [inline Sources/MapboxCoreNavigation/RouteStepProgress.swift:87]: Would it be possible to add unit-test that verifies such negative case when there are no intersections?
